core/src/test/java/org/elasticsearch/index/query/MissingQueryBuilderTest.java

```diff
@@ -0,0 +1,97 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.search.Query;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.is;
+
+public class MissingQueryBuilderTest extends BaseQueryTestCase<MissingQueryBuilder> {
+
+    @Override
+    protected MissingQueryBuilder doCreateTestQueryBuilder() {
+        MissingQueryBuilder query  = new MissingQueryBuilder(getRandomFieldName());
+        if (randomBoolean()) {
+            query.nullValue(randomBoolean());
+        }
+        if (randomBoolean()) {
+            query.existence(randomBoolean());
+        }
+        // cannot set both to false
+        if ((query.nullValue() == false) && (query.existence() == false)) {
+            query.existence(!query.existence());
+        }
+        return query;
+    }
+
+    private String getRandomFieldName() {
+        if (randomBoolean()) {
+            return randomAsciiOfLengthBetween(1, 10);
+        }
+        switch (randomIntBetween(0, 3)) {
+            case 0:
+                return BOOLEAN_FIELD_NAME;
+            case 1:
+                return STRING_FIELD_NAME;
+            case 2:
+                return INT_FIELD_NAME;
+            case 3:
+                return DOUBLE_FIELD_NAME;
+            default:
+                throw new UnsupportedOperationException();
+        }
+    }
+
+    @Override
+    protected Query doCreateExpectedQuery(MissingQueryBuilder queryBuilder, QueryParseContext context) throws IOException {
+        return MissingQueryBuilder.newFilter(context, queryBuilder.fieldPattern(), queryBuilder.existence(), queryBuilder.nullValue());
+    }
+
+    @Test
+    public void testValidate() {
+        MissingQueryBuilder missingQueryBuilder = new MissingQueryBuilder("");
+        assertThat(missingQueryBuilder.validate().validationErrors().size(), is(1));
+
+        missingQueryBuilder = new MissingQueryBuilder(null);
+        assertThat(missingQueryBuilder.validate().validationErrors().size(), is(1));
+
+        missingQueryBuilder = new MissingQueryBuilder("field").existence(false).nullValue(false);
+        assertThat(missingQueryBuilder.validate().validationErrors().size(), is(1));
+
+        missingQueryBuilder = new MissingQueryBuilder("field");
+        assertNull(missingQueryBuilder.validate());
+    }
+    
+    @Test
+    public void testBothNullValueAndExistenceFalse() throws IOException {
+        MissingQueryBuilder missingQueryBuilder = new MissingQueryBuilder("field").nullValue(false).existence(false);
+        QueryParseContext context = createContext();
+        context.setAllowUnmappedFields(true);
+        try {
+            missingQueryBuilder.toQuery(context);
```

Comment:
I would rather test this by calling the newFilter static method. In fact the check that we left there instead of relying on validate is there ony to protect other code paths from the outside (newFilter is public)